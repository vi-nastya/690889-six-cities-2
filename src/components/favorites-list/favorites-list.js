import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Operation} from "../../reducer";
import {getFavoritePlaces} from "../../selectors/selectors";

const FavoritesList = (props) => {
  const {favoritesData} = props;
  // need PureComponent, fetch data in componentDidMount
  return <div></div>;
};

FavoritesList.propTypes = {
  favoritesData: PropTypes.array.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    favoritesData: getFavoritePlaces(state),
  });
};

const mapDispatchToProps = (dispatch) => ({
  onLoadFavorites: () => {
    dispatch(Operation.loadFavorites());
  }
});

export {FavoritesList};

export default connect(mapStateToProps, mapDispatchToProps)(FavoritesList);
