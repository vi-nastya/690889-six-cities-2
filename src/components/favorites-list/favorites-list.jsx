import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Operation} from "../../reducer";
import {getGroupedFavoriteOffers} from "../../selectors/selectors";
import PlaceCard from "../place-card/place-card.jsx";
import Header from "../header/header.jsx";

class FavoritesList extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {onLoadFavorites} = this.props;
    onLoadFavorites();
  }

  render() {
    const {favoritesData} = this.props;

    if (favoritesData.length) {
      return <div>
        <div style={{display: `none`}}>
          <svg xmlns="http://www.w3.org/2000/svg">
            <symbol id="icon-arrow-select" viewBox="0 0 7 4"><path fillRule="evenodd" clipRule="evenodd" d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"></path></symbol>
            <symbol id="icon-bookmark" viewBox="0 0 17 18"><path d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"></path></symbol>
            <symbol id="icon-star" viewBox="0 0 13 12"><path fillRule="evenodd" clipRule="evenodd" d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"></path></symbol></svg>
        </div>

        <div className="page">
          <Header />
          <main className="page__main page__main--favorites">
            <div className="page__favorites-container container">
              <section className="favorites">
                <h1 className="favorites__title">Saved listing</h1>
                <ul className="favorites__list">
                  {favoritesData.map((placesGroup, index) =>
                    <li className="favorites__locations-items" key={`favorites-city-${index}`}>
                      <div className="favorites__locations locations locations--current">
                        <div className="locations__item">
                          <a className="locations__item-link" href="#">
                            <span>{placesGroup.city}</span>
                          </a>
                        </div>
                      </div>
                      <div className="favorites__places">
                        {placesGroup.offersList.map((placeData, placeIndex) =>
                          <PlaceCard cardClass={`favorites__card`} place={placeData} key={`favorite-place-${index}-${placeIndex}`}/>
                        )}
                      </div>
                    </li>
                  )}
                </ul>
              </section>
            </div>
          </main>
          <footer className="footer container">
            <a className="footer__logo-link" href="main.html">
              <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33"/>
            </a>
          </footer>
        </div>
      </div>;
    }

    return <div>
      <div style={{display: `none`}}>
        <svg xmlns="http://www.w3.org/2000/svg">
          <symbol id="icon-arrow-select" viewBox="0 0 7 4"><path fillRule="evenodd" clipRule="evenodd" d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"></path></symbol>
          <symbol id="icon-bookmark" viewBox="0 0 17 18"><path d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"></path></symbol>
          <symbol id="icon-star" viewBox="0 0 13 12"><path fillRule="evenodd" clipRule="evenodd" d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"></path></symbol></svg>
      </div>

      <div className="page page--favorites-empty">
        <Header/>
        <main className="page__main page__main--favorites page__main--favorites-empty">
          <div className="page__favorites-container container">
            <section className="favorites favorites--empty">
              <h1 className="visually-hidden">Favorites (empty)</h1>
              <div className="favorites__status-wrapper">
                <b className="favorites__status">Nothing yet saved.</b>
                <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
              </div>
            </section>
          </div>
        </main>
        <footer className="footer">
          <a className="footer__logo-link" href="main.html">
            <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33"/>
          </a>
        </footer>
      </div>
    </div>;
  }
}

FavoritesList.propTypes = {
  favoritesData: PropTypes.array.isRequired,
  onLoadFavorites: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    favoritesData: getGroupedFavoriteOffers(state),
  });
};

const mapDispatchToProps = (dispatch) => ({
  onLoadFavorites: () => {
    dispatch(Operation.loadFavorites());
  }
});

export {FavoritesList};

export default connect(mapStateToProps, mapDispatchToProps)(FavoritesList);
