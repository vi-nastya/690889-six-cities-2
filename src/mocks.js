export const MOCK_CITIES = [{
  name: `City1`,
  location: {
    latitude: 52.0,
    longitude: 4.5,
    zoom: 10
  }
},
{
  name: `City2`,
  location: {
    latitude: 20.4,
    longitude: 10.1,
    zoom: 3
  }
}
];

export const MOCK_USER = {
  id: 4,
  isPro: false,
  name: `Max`,
  avatarUrl: `img/1.png`
};

export const MOCK_OFFERS = [{
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  maxAdults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];

export const MOCK_OFFER_UPDATED_FAVORITE = {
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
};

export const MOCK_OFFERS_UPDATED_FAVORITE = [{
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  maxAdults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];


export const MOCK_REVIEWS = [{
  id: 1,
  user: MOCK_USER,
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
}];

export const MOCK_AUTH = {
  id: 4,
  isPro: false,
  name: `Max`,
  email: `max@test.com`,
  avatarUrl: `img/1.png`
};


/* eslint-disable camelcase */
export const MOCK_AUTH_SERVER = {
  id: 4,
  is_pro: false,
  name: `Max`,
  email: `max@test.com`,
  avatar_url: `img/1.png`
};


export const MOCK_USER_SERVER = {
  id: 4,
  is_pro: false,
  name: `Max`,
  avatar_url: `img/1.png`
};

export const MOCK_REVIEWS_SERVER = [{
  id: 1,
  user: MOCK_USER_SERVER,
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
}];


export const MOCK_OFFERS_SERVER = [{
  id: 1,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: false,
  is_premium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: true,
  is_premium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  max_adults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: false,
  is_premium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];

export const MOCK_OFFER_UPDATED_FAVORITE_SERVER = {
  id: 1,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: true,
  is_premium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
};
/* eslint-enable */
