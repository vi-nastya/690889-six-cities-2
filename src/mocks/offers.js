export const offers = [
  {
    id: 0,
    title: `Wood and stone place`,
    price: 80,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.3909553943508, longitude: 4.85309666406198},
    priceText: `night`,
    type: `Private room`,
    picture: `img/apartment-01.jpg`,
    isPremium: false,
    isFavorite: true,
    images: [`img/1.png`, `img/2.png`],
    bedrooms: 2,
    rating: 4,
    maxAdults: 2,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      is–óro: true,
      name: `Angelina`,
      avatarUrl: `img/1.png`
    },
  },
  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    price: 120,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.369553943508, longitude: 4.85309666406198},
    priceText: `night`,
    type: `Apartment`,
    picture: `img/apartment-02.jpg`,
    isPremium: true,
    isFavorite: false,
    rating: 3.6,
    images: [``]
  },
  {
    id: 2,
    title: `Nice, cozy, warm big bed apartment`,
    price: 180,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.3909553943508, longitude: 4.929309666406198},
    priceText: `night`,
    type: `Apartment`,
    picture: `img/apartment-03.jpg`,
    isPremium: true,
    isFavorite: true,
    images: [``],
    rating: 2,
  },
  {
    id: 3,
    title: `Wood and stone place`,
    price: 90,
    city: {name: `Moscow`, location: {latitude: 52.3809553943508, longitude: 4.939309666406198}},
    location: {latitude: 52.3809553943508, longitude: 4.939309666406198},
    priceText: `night`,
    type: `Private room`,
    picture: `https://picsum.photos/260/200`,
    isPremium: false,
    isFavorite: false,
    images: [``],
    rating: 4.8,
  },
];
