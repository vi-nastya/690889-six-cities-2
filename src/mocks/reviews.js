export const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatarUrl: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 1,
      isPro: true,
      name: `Mark`,
      avatarUrl: `img/2.png`
    },
    rating: 2,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-10T14:13:56.569Z`
  }
];
