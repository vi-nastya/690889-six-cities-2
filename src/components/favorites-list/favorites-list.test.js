import React from "react";
import renderer from "react-test-renderer";
import {FavoritesList} from "./favorites-list";
import {MOCK_OFFERS} from "../../mocks";

jest.mock(`../header/header`, () => jest.fn().mockReturnValue(null));
jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));

it(`FavoritesList component is rendered correctly after relaunch`, () => {
  const tree = renderer.create(<FavoritesList
    favoritesData={[{city: `Moscow`, offersList: MOCK_OFFERS}]}
    onLoadFavorites={jest.fn()}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
