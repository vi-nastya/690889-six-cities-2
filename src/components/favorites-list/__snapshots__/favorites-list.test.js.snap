// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList component is rendered correctly after relaunch 1`] = `
<div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>
  <div
    className="page"
  >
    <main
      className="page__main page__main--favorites"
    >
      <div
        className="page__favorites-container container"
      >
        <section
          className="favorites"
        >
          <h1
            className="favorites__title"
          >
            Saved listing
          </h1>
          <ul
            className="favorites__list"
          >
            <li
              className="favorites__locations-items"
            >
              <div
                className="favorites__locations locations locations--current"
              >
                <div
                  className="locations__item"
                >
                  <a
                    className="locations__item-link"
                    href="#"
                  >
                    <span>
                      Moscow
                    </span>
                  </a>
                </div>
              </div>
              <div
                className="favorites__places"
              >
                <article
                  className="favorites__card place-card"
                  onMouseEnter={null}
                  onMouseLeave={null}
                >
                  <div
                    className="place-card__mark"
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    className="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="#"
                    >
                      <img
                        alt="Place image"
                        className="place-card__image"
                        height="200"
                        src="img/1.png"
                        width="260"
                      />
                    </a>
                  </div>
                  <div
                    className="place-card__info"
                  >
                    <div
                      className="place-card__price-wrapper"
                    >
                      <div
                        className="place-card__price"
                      >
                        <b
                          className="place-card__price-value"
                        >
                          €
                          120
                        </b>
                        <span
                          className="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        className="place-card__bookmark-button button"
                        type="button"
                      >
                        <svg
                          className="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlinkHref="#icon-bookmark"
                          />
                        </svg>
                        <span
                          className="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      className="place-card__rating rating"
                    >
                      <div
                        className="place-card__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "100%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      className="place-card__name"
                    />
                    <p
                      className="place-card__type"
                    >
                      apartment
                    </p>
                  </div>
                </article>
                <article
                  className="favorites__card place-card"
                  onMouseEnter={null}
                  onMouseLeave={null}
                >
                  <div
                    className="place-card__mark"
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    className="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="#"
                    >
                      <img
                        alt="Place image"
                        className="place-card__image"
                        height="200"
                        src="img/1.png"
                        width="260"
                      />
                    </a>
                  </div>
                  <div
                    className="place-card__info"
                  >
                    <div
                      className="place-card__price-wrapper"
                    >
                      <div
                        className="place-card__price"
                      >
                        <b
                          className="place-card__price-value"
                        >
                          €
                          1
                        </b>
                        <span
                          className="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        className="place-card__bookmark-button button place-card__bookmark-button--active"
                        type="button"
                      >
                        <svg
                          className="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlinkHref="#icon-bookmark"
                          />
                        </svg>
                        <span
                          className="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      className="place-card__rating rating"
                    >
                      <div
                        className="place-card__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      className="place-card__name"
                    />
                    <p
                      className="place-card__type"
                    >
                      apartment
                    </p>
                  </div>
                </article>
                <article
                  className="favorites__card place-card"
                  onMouseEnter={null}
                  onMouseLeave={null}
                >
                  <div
                    className="place-card__mark"
                    style={Object {}}
                  >
                    <span>
                      Premium
                    </span>
                  </div>
                  <div
                    className="favorites__image-wrapper place-card__image-wrapper"
                  >
                    <a
                      href="#"
                    >
                      <img
                        alt="Place image"
                        className="place-card__image"
                        height="200"
                        src="img/1.png"
                        width="260"
                      />
                    </a>
                  </div>
                  <div
                    className="place-card__info"
                  >
                    <div
                      className="place-card__price-wrapper"
                    >
                      <div
                        className="place-card__price"
                      >
                        <b
                          className="place-card__price-value"
                        >
                          €
                          120
                        </b>
                        <span
                          className="place-card__price-text"
                        >
                          / night
                        </span>
                      </div>
                      <button
                        className="place-card__bookmark-button button"
                        type="button"
                      >
                        <svg
                          className="place-card__bookmark-icon"
                          height="19"
                          width="18"
                        >
                          <use
                            xlinkHref="#icon-bookmark"
                          />
                        </svg>
                        <span
                          className="visually-hidden"
                        >
                          To bookmarks
                        </span>
                      </button>
                    </div>
                    <div
                      className="place-card__rating rating"
                    >
                      <div
                        className="place-card__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "40%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <h2
                      className="place-card__name"
                    />
                    <p
                      className="place-card__type"
                    >
                      apartment
                    </p>
                  </div>
                </article>
              </div>
            </li>
          </ul>
        </section>
      </div>
    </main>
    <footer
      className="footer container"
    >
      <a
        className="footer__logo-link"
        href="main.html"
      >
        <img
          alt="6 cities logo"
          className="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;
